### PR TITLE
Feature: Allow blame to ignore whitespace change

### PR DESCRIPTION
**Pull request by ognarb**
Add a new blame flag GIT_BLAME_IGNORE_WHITESPACE and propagate it to xdiff.

Also, add a test for testing if correct behaviour is implemented and update the
blametest repo.

### CODE REVIEW

**Comment by Uncommon** [comment]
I just gotta say I enjoy the irony of that latest commit adjusting the whitespace in a change that enables ignoring whitespace changes :)

**Comment by ognarb** [comment]
Yes, I only saw that I was using space instead of tabs in the github pull request.

File: src/blame_git.c [hunk 1]
```diff
@@ -365,11 +365,14 @@ static void trim_common_tail(mmfile_t *a, mmfile_t *b, long ctx)
...
-static int diff_hunks(mmfile_t file_a, mmfile_t file_b, void *cb_data)
+static int diff_hunks(mmfile_t file_a, mmfile_t file_b, void *cb_data, git_blame_options *options)
 {
-	xpparam_t xpp = {0};
 	xdemitconf_t xecfg = {0};
 	xdemitcb_t ecb = {0};
+	xpparam_t xpp = {0 };
+	if (options->flags & GIT_BLAME_IGNORE_WHITESPACE) {
+		xpp.flags |= XDF_IGNORE_WHITESPACE;
+	}
```
Comment (pks-t) [inline src/blame_git.c]: Small style nit: we usually avoid braces around single-line statements in conditions. Please also remove the space in `{0 }` and add a newline between variables and first statement:

```
xpparam_t xpp = {0};

if (options->flags & GIT_BLAME_IGNORE_WHITESPACE)
        xpp.flags |= XDF_IGNORE_WHITESPACE;
```

Comment (ognarb) [reply src/blame_git.c]: Done and I rebased all my commits into one like you recommended me in IRC.

**Comment by pks-t** [comment]
On Sun, Feb 16, 2020 at 09:27:09AM -0800, Carl Schwan wrote:
> >  	xdemitconf_t xecfg = {0};
>  	xdemitcb_t ecb = {0};
> +	xpparam_t xpp = {0 };
> +	if (options->flags & GIT_BLAME_IGNORE_WHITESPACE) {
> +		xpp.flags |= XDF_IGNORE_WHITESPACE;
> +	}
> 
> Done and I rebased all my commits into one like you recommended me in IRC.

Perfect, thanks! I'll revisit this PR as soon as v1.0 is out.

**Comment by pks-t** [comment]
Oh, noticed just now. There's a merge conflict, so could you please rebase your commits on latest master and resolve the conflict?

**Comment by ognarb** [comment]
I now rebased. Sorry for the delay I was sick :(

**Comment by pks-t** [comment]
> I now rebased. Sorry for the delay I was sick :(

That's totally fine, no need to be sorry. Thanks for the rebase and your changes!

**Pull request merged**